Description: Fix alembic migration for SQLite 3
Author: Thomas Goirand <zigo@debian.org>
Forwarded: no
Last-Update: 2014-09-25

--- sahara-2014.2~b3.orig/sahara/db/migration/alembic_migrations/versions/007_increase_status_description_size.py
+++ sahara-2014.2~b3/sahara/db/migration/alembic_migrations/versions/007_increase_status_description_size.py
@@ -32,12 +32,18 @@ from sahara.db.sqlalchemy import types a
 
 
 def upgrade():
-    op.alter_column('clusters', 'status_description',
-                    type_=st.LongText(), existing_nullable=True,
-                    existing_server_default=None)
+    bind = op.get_bind()
+    engine = bind.engine
+    if engine.name != 'sqlite':
+        op.alter_column('clusters', 'status_description',
+                        type_=st.LongText(), existing_nullable=True,
+                        existing_server_default=None)
 
 
 def downgrade():
-    op.alter_column('clusters', 'status_description',
-                    type_=sa.String(length=200), existing_nullable=True,
-                    existing_server_default=None)
+    bind = op.get_bind()
+    engine = bind.engine
+    if engine.name != 'sqlite':
+        op.alter_column('clusters', 'status_description',
+                        type_=sa.String(length=200), existing_nullable=True,
+                        existing_server_default=None)
